source/puppet/5.5/lang_data_abstract.md

```diff
@@ -117,6 +117,72 @@ Note that you can use capture groups in the regular expressions, but they won't
+* An Integer or Float represents a number of seconds
+
+A `Timespan` range with either end defined as `default` (infinity) is an _open range_, while any other span is a _closed range_. The span's range is inclusive.
+
+The `Timespan` type is not enumerable.
+
+#### Examples:
+
+* `Timespan[2]` --- a timespan of 2 seconds
+* `Timespan[77.3]` --- a timespan of 1 minute, 17 seconds, and 300 milliseconds (77.3 seconds)
+* `Timespan['1-00:00:00', '2-00:00:00']` --- a timespan between 1 and 2 days
+
+### `Timestamp`
+
+The `Timestamp` data type matches a Timestamp; [strings][] or [numbers][] that represent a time and date; or `default`, which represents an infinite point in the past or future.
+
+It takes up to two parameters, and defaults to an infinite range to the past and future. A `Timestamp` with one parameter represents a single point in time, while two parameters represent a range of time, with the first parameter being the `from` value and the second being the `to` value.
+
+#### Parameters
+
+The full signature for `Timestamp` is:
+
+    Timestamp[ (<TIMESTAMP VALUE>, (<RANGE LIMIT>)) ]
+
+Position | Parameter | Data Type | Default Value | Description
+---------|-----------|-----------|---------------|------------
+1 | Timestamp Value | String, Float, Integer, or `default` | `default` (-Infinity in a range) | Point in time if passed alone, or `from` value in a range if passed with a second parameter
+2 | Range Limit | String, Float, Integer, or `default` | `default` (+Infinity), or none if only one value is passed | The `to` value in a range
+
+`Timestamp` values are interpreted depending on their format.
+
+* A String in the format of `YYYY-MM-DD` represents the time 00:00:00.0 UTC on the specified year (`YYYY`), month (`MM`), and day (`DD`)
+* A String in the format of `YYYY-MM-DD`T`HH:MM:SS.SSS` represents a specific hour (`HH`), minute (`MM`), and decimal second (`SS.SSS`) of UTC on the specified date
+* An Integer or Float represents the seconds passed since the Unix epoch
+* `default` is used in ranges and represents infinity as the maximum floating point value, either negative (as the `from` value of a range) or positive (as the `to` value)
+
+A `Timestamp` range with either end defined as `default` (infinity) is an _open range_, while a range with both ends defined as an exact timestamp is a _closed range_. Range values can be negative, and the range is inclusive.
+
+The `Timestamp` type is not enumerable.
+
+#### Examples:
+
+* `Timestamp['2000-01-01T00:00:00.000', default]` --- an open range of time from the start of the 21st century to an infinte point in the future
+* `Timestamp['2012-10-10']` --- the exact Timestamp 2012-10-10T00:00:00.0 UTC
+* `Timestamp[default, 1433116800]` --- an open range of time from an infinite point in the past to 2015-06-01T00:00:00 UTC, here expressed as seconds since the Unix epoch
```

Comment:
```suggestion
* `Timestamp[default, 1433116800]` --- an open range of timestamps from an infinite point in the past to 2015-06-01T00:00:00 UTC, here expressed as seconds since the Unix epoch
```